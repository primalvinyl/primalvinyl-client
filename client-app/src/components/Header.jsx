import React from 'react';
import { NavLink } from 'react-router-dom';
import { navList } from '../__types__';
import { useBreakpoint } from '../utilities/BreakpointProvider';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import './Header.scss';

const Header = () => {
    const breakpoint = useBreakpoint();
    const [toggleNav, setToggleNav] = React.useState(false);
    const NavList = navList.map(link => (
        <li key={link.path}>
            <NavLink exact to={link.path} key={link.path}>
                {link.value}
            </NavLink>
        </li>
    ));
    const toggleNavHandler = () => {
        setToggleNav(!toggleNav);
    };

    return (
        <header>
            <div className="wrapper">
                {breakpoint.lgDown &&
                    <button className="nav-toggle" onClick={toggleNavHandler}>
                    <FontAwesomeIcon icon={faBars} />
                    </button>
                }
                <a href="/" className="logotype">
                    <hgroup>
                        <h1>My Site</h1>
                        <h2>My Site</h2>
                    </hgroup>
                </a>
                {(toggleNav || breakpoint.lgUp) &&
                    <nav className="navigation">
                        <ul>{ NavList }</ul> 
                    </nav>
                }
            </div>
        </header>
    );
};

export default Header;
