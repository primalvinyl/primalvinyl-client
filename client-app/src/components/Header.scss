@import '../styles/partials/breakpoints';
@import '../styles/partials/variables';

header {
    padding: 2em;
    background: $gray-800;
    color: $gray-500;
    
    .wrapper {
        margin: 0 auto;
        display: flex;
        align-items: center;
        justify-content: space-between;

        @include breakpoint(lg-up){
            width: 1200px;
        }

        .logotype {
            text-decoration: none;

            hgroup {
                display: flex;
                align-items: center;

                h1 {
                    text-indent: -9999px;
                    width: 75px;
                    height: 75px;
                    background-image: url(../images/logo.svg);
                    background-position: center;
                    background-size: cover;
                    margin: 0 .5em 0 0;
                }

                h2 {
                    color: $gray-500;
                    font-weight: bold;
                    font-size: 2em;
                }
            }
        }

        .navigation {
            a {
                color: $gray-500;
                text-decoration: none;
                padding: 0 0 5px 0;
                &:hover {
                    color: $white;
                }
                &.active {
                    color: $white;
                    border-bottom: 1px solid $white;
                }
            }
            ul {
                display: flex;
                li {
                    margin: 0 0 0 2em;
                }
            }
        }
    }
}
