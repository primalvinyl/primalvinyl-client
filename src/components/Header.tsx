import React from 'react';
import { Link } from 'react-router-dom';
/*
import { NavLink, Link } from 'react-router-dom';
import { CSSTransition } from 'react-transition-group';
import { navList } from './types';
import { useBreakpoint } from '../utility/BreakpointProvider';
*/
import monster from '../images/monster.png';
import styles from './Header.module.scss';

const Header = (): React.ReactElement => {
    const headerRef = React.useRef<HTMLElement>(null);

    /*
    const breakpoint = useBreakpoint();
    const [toggleNav, setToggleNav] = React.useState(false);
    const NavList = navList.map(link => (
        <li key={link.path}>
            <NavLink exact to={link.path} key={link.path}>
                {link.value}
            </NavLink>
        </li> 
    ));

    // detects mouse clicks outside nav
    React.useEffect(() => {
        document.addEventListener('mousedown', clickHandler);
        return () => document.removeEventListener('mousedown', clickHandler);
    }, []);
 
    const clickHandler = (event: MouseEvent): void => {
        if (headerRef.current && headerRef.current.contains(event.target as Node)) return;
        else setToggleNav(false);
    };

    const toggleNavHandler = (event: React.MouseEvent): void => {
        event.stopPropagation();
        setToggleNav(!toggleNav);
    };
    */

    return (
        <header ref={headerRef} className={styles.root}>
            <div className={styles.wrapper}>
                {/*
                    breakpoint.lgDown &&
                    <button className={styles.navToggle} onClick={toggleNavHandler}>
                        <svg viewBox="0 0 100 80" width="25" height="30">
                            <rect width="100" height="15" rx="8" fill="gray"></rect>
                            <rect y="30" width="100" height="15" rx="8" fill="gray"></rect>
                            <rect y="60" width="100" height="15" rx="8" fill="gray"></rect>
                        </svg>
                    </button>
                */}
                <figure><img src={monster} alt="Logo" /></figure>
                <h1>
                    <Link to="/">
                        Primal Vinyl - Song Lyrics Search
                    </Link>
                </h1>
                {/*
                    <CSSTransition
                        in={toggleNav}
                        timeout={300}
                        classNames={{
                            enter: styles.enter,
                            enterDone: styles.enterDone
                    }}>
                        <nav className={styles.navigation}>
                            {(toggleNav || breakpoint.lgUp) &&
                                <ul>{NavList}</ul>
                            }
                        </nav>
                    </CSSTransition>
                */}
            </div>
        </header>
    );
};

export default Header;
