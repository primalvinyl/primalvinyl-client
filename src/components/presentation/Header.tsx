import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import { CSSTransition } from 'react-transition-group';
import { navList } from './types';
import { useBreakpoint } from '../utility/BreakpointProvider';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import styles from './Header.module.scss';

const Header = (): React.ReactElement => {
    const headerRef = React.useRef<HTMLElement>(null);
    const breakpoint = useBreakpoint();
    const [toggleNav, setToggleNav] = React.useState(false);
    const NavList = navList.map(link => (
        <li key={link.path}>
            <NavLink exact to={link.path} key={link.path}>
                {link.value}
            </NavLink>
        </li> 
    ));

    // detects mouse clicks outside nav
    React.useEffect(() => {
        document.addEventListener('mousedown', clickHandler);
        return () => document.removeEventListener('mousedown', clickHandler);
    }, []);
 
    const clickHandler = (event: MouseEvent): void => {
        if (headerRef.current && headerRef.current.contains(event.target as Node)) return;
        else setToggleNav(false);
    };

    const toggleNavHandler = (event: React.MouseEvent): void => {
        event.stopPropagation();
        setToggleNav(!toggleNav);
    };

    return (
        <header ref={headerRef} className={styles.root}>
            <div className={styles.wrapper}>
                {breakpoint.lgDown &&
                    <button className={styles.navToggle} onClick={toggleNavHandler}>
                        <FontAwesomeIcon icon={faBars} />
                    </button>
                }
                <Link to="/" className={styles.logotype}>
                    <hgroup>
                        <h1>Primal Vinyl</h1>
                        <h2>Primal Vinyl</h2>
                    </hgroup>
                </Link>
                <CSSTransition
                    in={toggleNav}
                    timeout={300}
                    classNames={{
                        enter: styles.enter,
                        enterDone: styles.enterDone
                }}>
                    <nav className={styles.navigation}>
                        {(toggleNav || breakpoint.lgUp) &&
                            <ul>{NavList}</ul>
                        }
                    </nav>
                </CSSTransition>
            </div>
        </header>
    );
};

export default Header;
