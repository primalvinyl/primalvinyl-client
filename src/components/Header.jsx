import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import { CSSTransition } from 'react-transition-group';
import { navList } from '../actions';
import { useBreakpoint } from '../utilities/BreakpointProvider';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import './Header.scss';

const Header = () => {
    const headerRef = React.useRef();
    const breakpoint = useBreakpoint();
    const [toggleNav, setToggleNav] = React.useState(false);
    const NavList = navList.map(link => (
        <li key={link.path}>
            <NavLink exact to={link.path} key={link.path}>
                {link.value}
            </NavLink>
        </li> 
    ));

    // detects mouse clicks outside the nav
    React.useEffect(() => {
        document.addEventListener('mousedown', clickHandler);
        return () => document.removeEventListener('mousedown', clickHandler);
    }, []);

    const clickHandler = event => {
        if (headerRef.current.contains(event.target)) return;
        else setToggleNav(false);
    };

    const toggleNavHandler = event => {
        event.stopPropagation();
        setToggleNav(!toggleNav);
    };

    return (
        <header ref={headerRef}>
            <div className="wrapper">
                {breakpoint.lgDown &&
                    <button className="nav-toggle" onClick={toggleNavHandler}>
                        <FontAwesomeIcon icon={faBars} />
                    </button>
                }
                <Link to="/" className="logotype">
                    <hgroup>
                        <h1>Primal Vinyl</h1>
                        <h2>Primal Vinyl</h2>
                    </hgroup>
                </Link>
                <CSSTransition in={toggleNav} timeout={300} classNames="nav-group">
                    <nav className="navigation">
                        {(toggleNav || breakpoint.lgUp) &&
                            <ul>{NavList}</ul>
                        }
                    </nav>
                </CSSTransition>
            </div>
        </header>
    );
};

export default Header;
