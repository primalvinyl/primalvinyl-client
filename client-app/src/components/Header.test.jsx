import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
    it('renders', () => {
        const myWrapper = shallow(<Header />);
        expect(myWrapper.exists());
    });
    it('matches snapshot', () => {
        const component = Renderer.create(<Header />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
