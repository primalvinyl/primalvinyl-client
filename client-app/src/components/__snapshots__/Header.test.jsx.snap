// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header>
  <div
    className="wrapper"
  >
    <a
      className="logotype"
      href="/"
    >
      <hgroup>
        <h1>
          My Site
        </h1>
        <h2>
          My Site
        </h2>
      </hgroup>
    </a>
  </div>
</header>
`;
